.github/workflows/build_and_test.yml

```diff
@@ -93,41 +83,56 @@ jobs:
     - name: Check labeler configuration
       run: scripts/check-labeler.pl .github/labeler.yml
 
-    - name: Unit tests
-      env:
-        KONG_TEST_PG_DATABASE: kong
-        KONG_TEST_PG_USER: kong
-      run: |
-          source ${{ env.BUILD_ROOT }}/kong-dev-venv.sh
-          TEST_CMD="bin/busted -v -o htest spec/01-unit"
-          if [[ $KONG_TEST_COVERAGE = true ]]; then
-            TEST_CMD="$TEST_CMD --coverage"
-          fi
-          $TEST_CMD
+  schedule:
+    name: Schedule busted tests to run
+    runs-on: ubuntu-22.04
+    needs: build
 
-    - name: Archive coverage stats file
+    env:
+      WORKFLOW_ID: ${{ github.run_id }}
+
+    outputs:
+      runners: ${{ steps.generate-runner-array.outputs.RUNNERS }}
+
+    steps:
+    - name: Checkout source code
+      uses: actions/checkout@v4
+
+    - name: Download runtimes file
+      uses: Kong/gh-storage/download@main
+      with:
+        repo-path: Kong/gateway-action-storage/main/.ci/runtimes.json
+
+    - name: Schedule tests
+      uses: Kong/gateway-test-scheduler/schedule@main
+      with:
+        test-suites-file: .ci/test_suites.json
+        test-file-runtime-file: .ci/runtimes.json
+        output-prefix: test-chunk.
+        runner-count: ${{ env.RUNNER_COUNT }}
+
+    - name: Upload schedule files
       uses: actions/upload-artifact@v3
-      if: ${{ always() && (inputs.coverage == true || github.event_name == 'schedule') }}
+      continue-on-error: true
       with:
-        name: luacov-stats-out-${{ github.job }}-${{ github.run_id }}
-        retention-days: 1
-        path: |
-          luacov.stats.out
+        name: schedule-test-files
+        path: test-chunk.*
+        retention-days: 7
 
-    - name: Get kernel message
-      if: failure()
+    - name: Generate runner array
+      id: generate-runner-array
       run: |
-        sudo dmesg -T
+        echo "RUNNERS=[$(echo $(seq 1 $(( $RUNNER_COUNT ))))]" | sed -e 's/ /, /g' >> $GITHUB_OUTPUT
 
-  integration-tests-postgres:
-    name: Postgres ${{ matrix.suite }} - ${{ matrix.split }} tests
+  busted-tests:
+    name: Busted test runner ${{ matrix.runner }}
```

Comment:
Currently this shows "Busted test runner 1", "Busted test runner 2", etc, is it possible to make this a bit more descriptive, like:

"Busted test runner 01/01/01-01/05/03"
"Busted test runner 01/05/04-02/03/03"
...

That can give me some hint on where one test might be located?

Reply:
The tests are run in the order of expected runtime, so there is no succinct way to put identifiers of the tests that are run into the name of the runner.  If you want to know what runner runs what test, you need to look into the output of the "Schedule tests" step of the "Schedule busted tests to run" Task in the ["Build & Test" workflow](https://github.com/Kong/kong/actions/runs/7214476053/job/19656684568):

<img width="1287" alt="image" src="https://github.com/Kong/kong/assets/108751/cf07994f-990f-45b4-953e-d4e9d1044b61">

Reply:
Discussed offline, we can merge as-is for now and improve later.